core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -307,6 +309,20 @@ public void registerSuggester(String key, Suggester<?> suggester) {
     }
 
     /**
+     * Lookup a {@link ScoreFunctionParser} by its name as used in XContent.
+     */
+    public ScoreFunctionParser<?> lookupScoreFunctionParser(String name, ParseFieldMatcher matcher) {
+        Tuple<ParseField, ScoreFunctionParser<?>> parser = scoreFunctionParsers.get(name);
+        if (parser == null) {
+            return null;
+        }
+        if (matcher.match(name, parser.v1())) {
+            return parser.v2();
+        }
+        return null;
```

Comment:
I would rather return an exception than null, or have asserts. If there is a parser, the parse field must match no? Also should we throw exception when the parser was not found? too much?